### PR TITLE
fix: handle nested alert keys

### PR DESCRIPTION
**Pull request by shahzad31**
Handle alert config in following format as well

```
heartbeat.monitors:
- type: http
  name: Todos Lightweight
  id: todos-lightweight
  enabled: true
  urls: "${devUrl}"
  schedule: '@every 3m'
  timeout: 16s
  alert:
    status.enabled: false
    tls.enabled: false
```

Edit: 

Continuation of https://github.com/elastic/synthetics/pull/785 which is related to [elastic/kibana#159697](https://github.com/elastic/kibana/pull/159697)

### CODE REVIEW

File: src/push/monitor.ts [hunk 1]
```diff
@@ -180,6 +180,7 @@ export async function createLightweightMonitors(
     const parsedDoc = parseDocument(content, {
       lineCounter,
       keepSourceTokens: true,
+      merge: true,
```
Comment (devcorpio) [inline src/push/monitor.ts:183]: Why is this change required?

Comment (shahzad31) [reply src/push/monitor.ts:183]: i will revert this. i was testing yaml parsing.

**Comment by devcorpio** [comment]
Should we support the following format, too?

```
  alert:
    status
      enabled: false
    tls
       enabled: false
```

File: src/push/monitor.ts (hunk 1 above)
Comment (shahzad31) [inline src/push/monitor.ts:183]:
```suggestion
```

**Comment by shahzad31** [comment]
> 

i think yes, let me add a use case for it. It's getting so ugly :D 

I wish yaml parser could handle it

**Review by vigneshshanmugam — CHANGES_REQUESTED** [review]
It would be better to add a solution to this problem for all keys instead of just doing for `alert`, this problem exists for all the nested fields in HB and its definitely not scalable solution to keep doing this for individual fields.

File: src/push/monitor.ts [hunk 2]
```diff
@@ -260,42 +260,49 @@ export function buildMonitorFromYaml(
...
   }
-  if (config['alert.tls.enabled'] !== undefined) {
-    const value = config['alert.tls.enabled'];
-    delete config['alert.tls.enabled'];
-    alertConfig['tls'] = {
-      enabled: value,
+  return Object.keys(result).length > 0 ? result : undefined;
+};
+
+export function getAlertKeyValue(
```
Comment (vigneshshanmugam) [inline src/push/monitor.ts:278]: can you add test for this which would take care of `delete behaviour`.

**Review by vigneshshanmugam — APPROVED** [review]
LGTM pending comments.
